src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(sep = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
+      }
+
+      if (i !== this.length - 1) {
+        result += this[i];
+        result += sep;
+      } else {
+        result += this[i];
```

Comment:
Don't repeat yourself (result += this[i]): you can move it outside of if / else statement